src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -0,0 +1,341 @@
+export class OutlinePerKeyDataLimitDialog extends LitElement {
+  /** @member key The UI access key representing the key we're working on. */
+  @internalProperty() key: DisplayAccessKey = null;
+  /** @member key The default data limit of the server we're working on, or null */
+  @internalProperty() serverDefaultLimit: DisplayDataAmount = null;
+  /**
+   * @member showMenu Whether the menu for inputting the data limit should be shown. Controlled by
+   * the checkbox.
+   */
+  @internalProperty() showMenu = false;
+  /**
+   * @member enableSave Whether the save button is enabled.  Controlled by the validator on the
+   * input.
+   */
+  @internalProperty() enableSave = false;
+  @property({type: Function}) localize: Function;
+
+  static get styles() {
+    return [
+      COMMON_STYLES,
+      css`
+        #container {
+          width: 100%;
+          display: flex;
+          flex-flow: column nowrap;
+        }
+
+        #dataLimitIcon {
+          /* Split the padding evenly between the icon and the section to be bidirectional. */
+          padding: 0 12px;
+        }
+
+        #headerSection {
+          display: flex;
+          flex-direction: row;
+          padding: 0 12px;
+        }
+
+        #headerSection h3 {
+          font-size: 18px;
+          font-weight: 500;
+          color: rgba(0, 0, 0, 0.87);
+          line-height: 24px;
+        }
+
+        #menuSection {
+          flex: 1;
+          padding: 0 78px;
+          margin-top: 10px;
+        }
+
+        #buttonsSection {
+          margin-top: 10px;
+          display: flex;
+          flex-direction: row-reverse;
+        }
+
+        paper-button {
+          display: flex;
+          height: 36px;
+          text-align: center;
+        }
+
+        #save {
+          background-color: var(--primary-green);
+          color: #fff;
+        }
+
+        #save[disabled] {
+          color: var(--dark-gray);
+          background-color: rgba(0, 0, 0, 0.13);
+        }
+
+        #menu {
+          display: flex;
+          flex-flow: row nowrap;
+        }
+
+        #unitsDropdown {
+          width: 50px;
+          padding: 0 10px;
+        }
+
+        paper-checkbox {
+          /* We want the ink to be the color we're going to, not coming from */
+          --paper-checkbox-checked-color: var(--primary-green);
+          --paper-checkbox-checked-ink-color: var(--dark-gray);
+          --paper-checkbox-unchecked-color: var(--dark-gray);
+          --paper-checkbox-unchecked-ink-color: var(--primary-green);
+        }
+
+        paper-listbox paper-item:hover {
+          cursor: pointer;
+          background-color: var(--background-contrast-color);
+          color: #fff;
+        }
+      `,
+    ];
+  }
+
+  render() {
+    // this.key will always be defined once the dialog is open, but before it's opened we get an
+    // error if we don't account for the undefined key
+    const keyName = this.key?.name || this.key?.placeholderName;
```

Comment:
Make this a property instead.

Reply:
Done in c214d2a67bdc82a0c76d1542c9d6b49c07a580b0